src/Features/CSharp/Portable/CodeFixes/ConditionalExpressionInStringInterpolation/CSharpAddParenthesesAroundConditionalExpressionInInterpolatedStringCodeFixProvider.cs

```diff
@@ -0,0 +1,80 @@
+
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(CS8361);
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var diagnostic = context.Diagnostics.First();
+            var diagnosticSpan = diagnostic.Location.SourceSpan;
+            var token = root.FindToken(diagnosticSpan.Start);
+            var conditionalExpression = token.GetAncestor<ConditionalExpressionSyntax>();
+            if (conditionalExpression != null)
+            {
+                var documentChangeAction = new MyCodeAction(
+                    cancellationToken => GetChangedDocumentAsync(context.Document, conditionalExpression.SpanStart, cancellationToken));
+                context.RegisterCodeFix(documentChangeAction, diagnostic);
+            }
+        }
+
+        private static async Task<Document> GetChangedDocumentAsync(Document document, int conditionalExpressionSyntaxStartPosition, CancellationToken cancellationToken)
+        {
+            // The usual SyntaxTree transformations are complicated if string literals are present in the false part as in
+            // $"{ condition ? "Success": "Failure" }"
+            // The colon starts a FormatClause and the double quote left to 'F' therefore ends the interpolated string.
+            // The text starting with 'F' is parsed as code and the resulting syntax tree is impractical.
+            // The same problem arises if a } is present in the false part.
+            // To circumvent these problems this solution
+            // 1. Inserts an opening parenthesis
+            // 2. Re-parses the resulting document (now the colon isn't treated as starting a FormatClause anymore)
+            // 3. Replaces the missing CloseParenToken with a new one
+            var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            var openParenthesisPosition = conditionalExpressionSyntaxStartPosition;
+            var textWithOpenParenthesis = text.Replace(openParenthesisPosition, 0, "(");
+            var documentWithOpenParenthesis = document.WithText(textWithOpenParenthesis);
+            var syntaxRoot = await documentWithOpenParenthesis.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var nodeAtInsertPosition = syntaxRoot.FindNode(new TextSpan(openParenthesisPosition, 0));
+            if (nodeAtInsertPosition is ParenthesizedExpressionSyntax parenthesizedExpression &&
+                parenthesizedExpression.CloseParenToken.IsMissing)
+            {
+                var parenthesizedExpressionWithClosingParen =
+                    parenthesizedExpression.WithCloseParenToken(SyntaxFactory.Token(SyntaxKind.CloseParenToken)).
+                    WithTriviaFrom(parenthesizedExpression);
```

Comment:
this is slightly odd trivia-wise.  likely better  to have:  parenthesizedExpression.WithoutTrivia().WithClose(...).WithTriviaFrom(parenthesized).

This prevents a doubling up of trivia which is technically possible here.

Reply:
I see. That's why I had problems with some disappearing whitespace. I fixed it slightly different (see [here](https://github.com/dotnet/roslyn/pull/23587/commits/d56ce9eb343264fbaa6bbf7e8e2ab44f878d479f#diff-9a6996bd4fc7eaf5b2a0be89c39f1255R59)).